xhyve/xhyve.go

```diff
@@ -603,22 +603,21 @@ func (d *Driver) extractKernelImages() error {
 	}
 
 	volumeRootDir := d.ResolveStorePath(isoMountPath)
-	vmlinuz := filepath.Join(volumeRootDir, "boot", d.Vmlinuz)
-	initrd := filepath.Join(volumeRootDir, "boot", d.Initrd)
 
-	log.Debugf("Extracting %s into %s", d.Vmlinuz, d.ResolveStorePath(d.Vmlinuz))
-	if err := mcnutils.CopyFile(vmlinuz, d.ResolveStorePath(d.Vmlinuz)); err != nil {
-		return err
-	}
-
-	log.Debugf("Extracting %s into %s", d.Initrd, d.ResolveStorePath(d.Initrd))
-	if err := mcnutils.CopyFile(initrd, d.ResolveStorePath(d.Initrd)); err != nil {
-		return err
-	}
+	defer func() {
+		log.Debugf("Unmounting %s", isoFilename)
+		err = hdiutil("detach", volumeRootDir)
+	}()
 
-	log.Debugf("Unmounting %s", isoFilename)
-	if err := hdiutil("detach", volumeRootDir); err != nil {
-		return err
+	for _, f := range []string{"vmlinuz64", "bzImage", "initrd", "initrd.img"} {
```

Comment:
for feature use, I think to add vmlinuz(generally compressed kernel image name) and vmlinux(not compressed) is better.
https://en.wikipedia.org/wiki/Vmlinux
> Traditionally, UNIX platforms called the kernel image /unix. With the development of virtual memory, kernels that supported this feature were given the vm- prefix to differentiate them. The name vmlinux is a mutation of vmunix, while in vmlinuz the letter z at the end denotes that it is compressed (gzipped).[1]

What do you think?